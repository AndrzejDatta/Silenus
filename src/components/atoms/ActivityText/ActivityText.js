import React from "react";
import PropTypes from "prop-types";
import { StyledActivityText } from "./ActivityText.styles";
const ActivityText = ({ text, isHeadline, color }) => {
  return (
    <StyledActivityText isHeadline={isHeadline} color={color}>
      {text}
    </StyledActivityText>
  );
};
ActivityText.defaultProps = {
  text: "Text",
  isHeadline: false,
  color: "black",
};
ActivityText.propTypes = {
  /**text in element */
  text: PropTypes.string.isRequired,
  /** bool which decides about font-weight and font-size */
  isHeadline: PropTypes.bool,
  /**color of font */
  color: PropTypes.string,
};

export default ActivityText;
