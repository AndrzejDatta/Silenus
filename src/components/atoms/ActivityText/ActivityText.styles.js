import styled from "styled-components";

export const StyledActivityText = styled.p`
  letter-spacing: 1px;
  font-size: ${(props) =>
    props.isHeadline
      ? ({ theme }) => theme.font.size.big
      : ({ theme }) => theme.font.size.medium};
  color: ${(props) => props.color};
  font-family: ${({ theme }) => theme.font.family.poppins};
  font-weight: ${(props) => (props.isHeadline ? "bold" : "normal")};
`;
