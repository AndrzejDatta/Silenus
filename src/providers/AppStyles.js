import { ThemeProvider } from 'styled-components';
import { theme } from 'assets/styles/theme';
import { GlobalStyle } from 'assets/styles/GlobalStyles';
export const AppStyles = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      {children}
    </ThemeProvider>
  );
};
