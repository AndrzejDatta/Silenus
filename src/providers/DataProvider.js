import { createContext, useState } from "react";
import { dataPL } from "assets/data/data";

export const dataContext = createContext({
  startPage: {},
  loginPage: {},
  Dashboard: {},
  MojeDoniczki: {},
  DodajDoniczke: {},
  MojeRośliny: {},
  KalendarzOpieki: {},
  KontoUżytkownika: {},
  ZidentyfikujRośline: {},
  PotDetails: {},
  PlantDetails: {},
  Navbar: {},
  images: {},
  navIcons: {},
  potIcons: {},
  categoryIcons: {},
  navIconsColor: [],
  setNavIconColor: () => {},
  changeColor: () => {},
  importAll: () => {},
});

const objectName = [
  "startPage",
  "loginPage",
  "Dashboard",
  "MojeDoniczki",
  "DodajDoniczke",
  "MojeRośliny",
  "KalendarzOpieki",
  "KontoUżytkownika",
  "ZidentyfikujRośline",
  "PotDetails",
  "PlantDetails",
  "Navbar",
];

export const DataProvider = ({ children }) => {
  for (let i = 0; i < dataPL.length; i++) {
    objectName[i] = { ...dataPL[i] };
  }
  const [
    startPage,
    loginPage,
    Dashboard,
    MojeDoniczki,
    DodajDoniczke,
    MojeRośliny,
    KalendarzOpieki,
    KontoUżytkownika,
    ZidentyfikujRośline,
    PotDetails,
    PlantDetails,
    Navbar,
  ] = objectName;

  const [navIconsColor, setNavIconColor] = useState([
    {
      name: "plant",
      src: "potted_plant.svg",
      isClicked: false,
    },
    {
      name: "calendar",
      src: "calendar.svg",
      isClicked: false,
    },
    {
      name: "camera",
      src: "camera.svg",
      isClicked: false,
    },
    {
      name: "user",
      src: "accountCircle.svg",
      isClicked: false,
    },
  ]);

  function importAll(r) {
    let images = {};
    r.keys().map((item, index) => {
      images[item.replace("./", "")] = r(item);
    });
    return images;
  }

  const images = importAll(
    require.context(
      "assets/icons/userAccountOption",
      false,
      /\.(png|jpe?g|svg)$/
    )
  );
  const navIcons = importAll(
    require.context("assets/icons/navIcons", false, /\.(png|jpe?g|svg)$/)
  );

  const potIcons = importAll(
    require.context("assets/icons/potStatus", false, /\.(png|jpe?g|svg)$/)
  );

  const categoryIcons = importAll(
    require.context("assets/icons/categoryIcons", false, /\.(png|jpe?g|svg)$/)
  );

  function changeColor(nameElement) {
    setNavIconColor((prevData) =>
      prevData.map((item) => {
        if (item.name === nameElement && !item.isClicked) {
          return {
            ...item,
            isClicked: true,
          };
        } else {
          return {
            ...item,
            isClicked: false,
          };
        }
      })
    );
  }

  return (
    <dataContext.Provider
      value={{
        startPage,
        loginPage,
        Dashboard,
        DodajDoniczke,
        MojeDoniczki,
        MojeRośliny,
        KalendarzOpieki,
        KontoUżytkownika,
        ZidentyfikujRośline,
        PotDetails,
        PlantDetails,
        Navbar,
        images,
        navIcons,
        potIcons,
        categoryIcons,
        navIconsColor,
        setNavIconColor,
        changeColor,
        importAll,
      }}
    >
      {children}
    </dataContext.Provider>
  );
};

//TODO:improve way
