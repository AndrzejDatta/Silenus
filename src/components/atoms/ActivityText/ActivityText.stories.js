import React from "react";
import { storiesName } from "helpers/storiesName";
import ActivityText from "./ActivityText";

export default {
  title: `${storiesName.atoms}ActivityText`,
  component: ActivityText,
  argTypes: {},
};

const Template = (args) => <ActivityText {...args} />;

export const Default = Template.bind({});

Default.args = {
  text: "Text",
  isHeadline: false,
  color: "black",
};
