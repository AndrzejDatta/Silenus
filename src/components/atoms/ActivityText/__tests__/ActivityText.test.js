import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { AppStyles } from "providers/AppStyles";
import ActivityText from "../ActivityText";
import { render, screen, rerender } from "@testing-library/react";

describe("Test ActivityText component", () => {
  it("snapshot - default style", () => {
    const tree = renderer
      .create(
        <AppStyles>
          <ActivityText />
        </AppStyles>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule("font-weight", "normal");
    expect(tree).toHaveStyleRule("color", "black");
    expect(tree).toHaveStyleRule("font-size", "14px");
  });
  it("snapshot - headline", () => {
    const tree = renderer
      .create(
        <AppStyles>
          <ActivityText isHeadline={true} />
        </AppStyles>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule("font-weight", "bold");
    expect(tree).toHaveStyleRule("font-size", "16px");
  });

  // it("test render text", () => {
  //   render(
  //     <AppStyles>
  //       <ActivityText text="Test" />
  //     </AppStyles>
  //   );

  //   screen.debug();
  //   const component = screen.getByText(/test/i);
  //   expect(component).toHaveStyle(`color: "black" `);
  // });
});
