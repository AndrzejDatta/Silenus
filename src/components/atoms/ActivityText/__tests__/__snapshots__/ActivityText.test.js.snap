// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ActivityText component snapshot - default style 1`] = `
.c0 {
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 14px;
  color: black;
  font-family: "Poppins",sans-serif;
  font-weight: normal;
}

<p
  className="c0"
  color="black"
>
  Text
</p>
`;

exports[`Test ActivityText component snapshot - headline 1`] = `
.c0 {
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 16px;
  color: black;
  font-family: "Poppins",sans-serif;
  font-weight: bold;
}

<p
  className="c0"
  color="black"
>
  Text
</p>
`;
